core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackend.scala

```diff
@@ -111,26 +115,20 @@ private[spark] class MesosSchedulerBackend(
       command.setValue(s"cd ${basename}*; $prefixEnv ./bin/spark-class $executorBackendName")
       command.addUris(CommandInfo.URI.newBuilder().setValue(uri.get))
     }
-    val cpus = Resource.newBuilder()
-      .setName("cpus")
-      .setType(Value.Type.SCALAR)
-      .setScalar(Value.Scalar.newBuilder()
-        .setValue(mesosExecutorCores).build())
-      .build()
-    val memory = Resource.newBuilder()
-      .setName("mem")
-      .setType(Value.Type.SCALAR)
-      .setScalar(
-        Value.Scalar.newBuilder()
-          .setValue(MemoryUtils.calculateTotalMemory(sc)).build())
-      .build()
-    MesosExecutorInfo.newBuilder()
+    val builder = MesosExecutorInfo.newBuilder()
+    val (resourcesAfterCpu, usedCpuResources) =
+      partitionResources(resources, "cpus", scheduler.CPUS_PER_TASK)
+    builder.addAllResources(usedCpuResources)
+    val (resourcesAfterMem, usedMemResources) =
+      partitionResources(resourcesAfterCpu, "mem", MemoryUtils.calculateTotalMemory(sc))
+    builder.addAllResources(usedMemResources)
+    val executorInfo = builder
```

Comment:
super minor, but this chunk will be much easier to read if you reorganize it a little:
```
val (resourcesAfterCpu, usedCpuResources) = ...
val (resourcesAfterMem, usedMemResources) = ...
val executorInfo = MesosExecutorInfo.newBuilder()
  .setExecutorId(...)
  ...
  .addAllResources(usedCpuResources)
  .addAllResources(usedMemResources)
  .build()

(executorInfo, resourcesAfterMem)
```